File: spp/services/go/daemon_service.go
package main

import (
	"fmt"
	"time"

	"github.com/spp/polyglot"
)

func main() {
	fmt.Println("Simulating heavy Go startup (takes 2 seconds)...")
	time.Sleep(2 * time.Second)
	fmt.Println("Go module initialized.")

	polyglot.Serve(dispatch)
}

func dispatch(funcName string, args []interface{}) interface{} {
	if funcName == "generate" {
		prompt, ok := args[0].(string)
		if !ok {
			prompt = "Unknown"
		}
		return "Go AI says: Hello! You asked: " + prompt
	}
	return map[string]string{"error": "Function not found"}
}

File: spp/services/java/DaemonService.java
package services.java;

import spp.polyglot.PolyglotBridge;
import spp.polyglot.PolyglotBridge.Dispatcher;

public class DaemonService {
    public static void main(String[] args) throws InterruptedException {
        System.out.println("Simulating heavy Java startup (takes 2 seconds)...");
        Thread.sleep(2000);
        System.out.println("Java module initialized.");

        Dispatcher dispatch = (funcName, argsJson) -> {
            if ("generate".equals(funcName)) {
                String prompt = "Unknown";
                String json = (String)argsJson;
                if (json.startsWith("[") && json.endsWith("]")) {
                    String inner = json.substring(1, json.length() - 1);
                    if (inner.startsWith("\"") && inner.endsWith("\"")) {
                        prompt = inner.substring(1, inner.length() - 1);
                    }
                }
                return "Java AI says: Hello! You asked: " + prompt;
            }
            return "{\"error\": \"Function not found\"}";
        };

        PolyglotBridge.serve(args, dispatch);
    }
}
